### PR TITLE
Add projected_enactment_at_start_of_epoch to pending protocol update in health API

### PR DESCRIPTION
**Pull request by LukasGasior1**
Adds (all optional):
- `projected_fulfillment_at_start_of_epoch` and `projected_fulfillment_timestamp` for each threshold of a pending protocol update
- `projected_enactment_at_start_of_epoch` and `projected_enactment_timestamp` to a pending protocol update, which are the values of the threshold which is projected to fulfill earliest

Notes:
- if the threshold is not met then its values are null
- if projected epoch is out of min/max bounds then the result is also null

### CODE REVIEW

File: core/src/main/java/com/radixdlt/monitoring/LedgerSummary.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * permissions under this License.
+ */
+
+package com.radixdlt.monitoring;
+
+import com.radixdlt.statecomputer.ProtocolState;
+import com.radixdlt.statecomputer.commit.ActiveValidatorInfo;
+import java.util.Set;
+
+public record LedgerSummary(
```
Comment (jakrawcz-rdx) [inline core/src/main/java/com/radixdlt/monitoring/LedgerSummary.java:71]: This does not "summarize ledger" (a broad concept), but it looks very specific to the protocol update and its tracking/forecasting, and hence deserves longer name :p
(I mean: let's keep this nice, concise "LedgerSummary" for a more appropriate occasion)
"ProtocolUpdateTrackingContext"?

File: core/src/test/java/com/radixdlt/api/system/HealthHandlerTest.java [hunk 2]
```diff
@@ -64,17 +64,97 @@
...
+  // Test assertion parameters
+  private static final long EXPECTED_PROJECTED_ENACTMENT_EPOCH = 116;
+  private static final long EXPECTED_PROJECTED_ENACTMENT_TIMESTAMP =
+      EPOCH_EFFECTIVE_START_MS + (EPOCH_TARGET_DURATION_MS * 16);
+
   @Inject private HealthHandler sut;
 
+  public HealthHandlerTest() {
+    super(
+        new AbstractModule() {
```
Comment (jakrawcz-rdx) [inline core/src/test/java/com/radixdlt/api/system/HealthHandlerTest.java:115]: The `Modules.override()` and the mocks here look rather cumbersome for testing :(
At the same time, I agree that the "forecast" calculations are worth testing.

How about you extract the calculations out from `*Handler`, into a specialized `*Forecaster` class, and cover it with a unit test instead?

File: core/src/main/java/com/radixdlt/monitoring/LedgerSummary.java (hunk 1 above)
Comment (dhedey) [reply core/src/main/java/com/radixdlt/monitoring/LedgerSummary.java:71]: Agreed with `LedgerSummary` not being a great name. Maybe `CurrentEpochParameters`, as I see it's used in the `MetricInstaller`?

File: core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java [hunk 3]
```diff
@@ -122,24 +130,85 @@ public HealthResponse handleRequest() {
...
+      LedgerSummary ledgerSummary) {
+    final var currentEpoch = ledgerSummary.currentEpoch();
+    final var epochStartTimeCalculator =
+        new EpochStartTimeCalculator(
+            ledgerSummary.consensusManagerConfigEpochTargetDurationMs(),
+            currentEpoch,
+            ledgerSummary.consenusManagerStateEpochEffectiveStartMs());
+
+    return switch (pendingProtocolUpdate.protocolUpdateTrigger().enactmentCondition()) {
+      case EnactAtStartOfEpochIfValidatorsReady condition -> {
```
Comment (dhedey) [inline core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java:150]: To make it easier to verify line-by-line, I'd rather this were called `atStartOfEpochCondition`

File: core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java [hunk 4]
```diff
@@ -122,24 +130,85 @@ public HealthResponse handleRequest() {
...
+                              thresholdState.last().consecutiveStartedEpochsOfSupport().toLong();
+                          if (consecutiveStartedEpochsOfSupport <= 0) {
+                            return Option.<Long>empty();
+                          } else {
+                            final var requiredConsecutiveCompletedEpochsOfSupport =
+                                thresholdState
+                                    .first()
+                                    .requiredConsecutiveCompletedEpochsOfSupport()
+                                    .toLong();
+                            final var projectedFulfillmentEpoch =
```
Comment (dhedey) [inline core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java:169]: To make it easier to verify line-by-line, I'd prefer this to be called `projectedFulfillmentAtStartOfEpoch`

**Review by dhedey — APPROVED** [review]
Really nice, thanks 👍. Will save us second-guessing ourselves when live.

File: core/src/main/java/com/radixdlt/monitoring/LedgerSummary.java (hunk 1 above)
Comment (LukasGasior1) [reply core/src/main/java/com/radixdlt/monitoring/LedgerSummary.java:71]: I might just go with `InMemorySystemInfoState`

**Pull request merged**